src/Training.AirBnb.Clone.Backend/Backend Project/Infrastructure/CompositionServices/LocationManagementService.cs

```diff
@@ -0,0 +1,182 @@
+﻿using Backend_Project.Application.Foundations.LocationServices;
+using Backend_Project.Application.Locations.Services;
+using Backend_Project.Domain.Entities;
+using Backend_Project.Domain.Exceptions.EntityExceptions;
+using Backend_Project.Domain.Extensions;
+
+namespace Backend_Project.Infrastructure.CompositionServices;
+
+public class LocationManagementService : ILocationManagementService
+{
+    private readonly IAddressService _addressService;
+    private readonly ILocationService _locationService;
+    private readonly IScenicViewService _scenicViewService;
+    private readonly ILocationScenicViewsService _locationScenicViewsService;
+    private readonly ICityService _cityService;
+
+    public LocationManagementService(IAddressService addressService, ILocationService locationService, IScenicViewService scenicViewService, ILocationScenicViewsService locationScenicViewsService, ICityService cityService)
+    {
+        _addressService = addressService;
+        _locationService = locationService;
+        _scenicViewService = scenicViewService;
+        _locationScenicViewsService = locationScenicViewsService;
+        _cityService = cityService;
+    }
+
+    public async ValueTask<Location> CreateLocationByAddressId(Guid addressId)
+    {
+        _ = await _addressService.GetByIdAsync(addressId);
+
+        var location = await _locationService.CreateAsync(new Location { AddressId = addressId });
+
+        return location;
+    }
+
+    public async ValueTask<Location> UpdateLocation(Location location)
+    {
+        await _addressService.GetByIdAsync(location.AddressId);
+
+        var updatedLocation = await _locationService.UpdateAsync(location);
+
+        return updatedLocation;
+    }
+
+    public async ValueTask<Location> DeleteLocationById(Guid locationId)
+    {
+        var deletedLocation = await _locationService.GetByIdAsync(locationId);
+
+        var locationScenicViews = _locationScenicViewsService
+            .Get(self => true)
+            .Where(locViews => locViews.LocationId == locationId)
+            .ToList();
+
+        var addresses = _addressService
```

Comment:
Address bilan location one to one relationship. Shuning uchun list qilib address lar qaytmidi. Bittagina address qaytadi shuni o'chirib yuborish kerak.